### DISCUSSION TITLE
Resolving File Search Issues: Handling Special Characters and Ensuring Correct URL Encoding

### DISCUSSION THREAD

**Discussion by azizul939** [prasathmani/tinyfilemanager#1044] [Show and tell]
I noticed that clicking the link in the search results for a folder or file with special characters would not display the file even if it exists. To address this issue, I made the following adjustments to the code:

// Search template
   function search_template(data) {
    var response = "";
    $.each(data, function (key, val) {
        var encodedPath = encodeURIComponent(val.path);
        var encodedName = encodeURIComponent(val.name);
        var url = `?p=${encodedPath}&view=${encodedName}`;
        response += `<li><a href="${url}">${val.path}/${val.name}</a></li>`;
    });
    return response;
}

In this code, I resolved the issue by using the encodeURIComponent() function separately for the val.path and val.name variables. This ensures that special characters are properly encoded before concatenating them in the URL. As a result, the search links now direct to the correct file, even if it contains special characters.

**Comment by prasathmani** [comment]
@azizul939 thanks for sharing, can you create a PR. it will be useful for others.
